src/DynamoCore/Updates/UpdateManager.cs

```diff
@@ -37,24 +37,98 @@ public UpdateDownloadedEventArgs(Exception error, string fileLocation)
     /// </summary>
     public interface IUpdateManager
     {
+        /// <summary>
+        /// Current product version.
+        /// </summary>
         BinaryVersion ProductVersion { get; }
+
+        /// <summary>
+        /// Available product version.
+        /// </summary>
         BinaryVersion AvailableVersion { get; }
+
+        /// <summary>
+        /// Information, where version can be updated.
+        /// </summary>
         IAppVersionInfo UpdateInfo { get; set; }
+
+        /// <summary>
+        /// Bool value indicates if new version is available.
+        /// </summary>
         bool IsUpdateAvailable { get; }
+
+        /// <summary>
+        /// Event fired, when update is downloaded.
+        /// </summary>
         event UpdateDownloadedEventHandler UpdateDownloaded;
+
+        /// <summary>
+        /// Event fired, when Dynamo needs to be restarted.
+        /// </summary>
         event ShutdownRequestedEventHandler ShutdownRequested;
+
+        /// <summary>
+        /// Checks for product updates in background thread.
+        /// </summary>
+        /// <param name="request">Asynchronous web request for update data</param>
         void CheckForProductUpdate(IAsynchronousRequest request);
+
+        /// <summary>
+        /// Quits and installs new version.
+        /// </summary>
         void QuitAndInstallUpdate();
+
+        /// <summary>
+        /// Called, when Dynamo Model is shuting down.
+        /// </summary>
         void HostApplicationBeginQuit();
+
+        /// <summary>
+        /// Reads the request's data, and parses for available versions. 
+        /// If a more recent version is available, the UpdateInfo object 
+        /// will be set.
+        /// </summary>
+        /// <param name="request">Asynchronous request</param>
         void UpdateDataAvailable(IAsynchronousRequest request);
+
+        /// <summary>
+        /// This flag is available via the debug menu to
+        /// allow the update manager to check for newer daily 
+        /// builds as well.
+        /// </summary>
         bool CheckNewerDailyBuilds { get; set; }
+
+        /// <summary>
+        /// Defines whether to force update, default value is false.
+        /// </summary>
         bool ForceUpdate { get; set; }
+
+        /// <summary>
+        /// Configurations.
```

Comment:
Returns a reference to Configuration...
see, in this case it is an interface. so specifty CREF here.